mne/io/meas_info.py

```diff
@@ -125,6 +125,28 @@ def _stamp_to_dt(stamp):
             datetime.timedelta(0, 0, stamp[1]))  # day, sec, μs
 
 
+def _unique_channel_names(ch_names):
+    """Ensure unique channel names."""
+    # refactored and modified from Info._check_consistency()
+    unique_ids = np.unique(ch_names, return_index=True)[1]
+    if len(unique_ids) != len(ch_names):
+        dups = set(ch_names[x]
+                   for x in np.setdiff1d(range(len(ch_names)), unique_ids))
+        warn('Channel names are not unique, found duplicates for: '
+             '%s. Applying running numbers for duplicates.' % dups)
+        for ch_stem in dups:
+            overlaps = np.where(np.array(ch_names) == ch_stem)[0]
+            n_keep = min(len(ch_stem),
+                         14 - int(np.ceil(np.log10(len(overlaps)))))
```

Comment:
yes 14 should be a `NICE_NAME = 14` constant at the top of the function. 

BTW, @cbrnr do you know why is it 14?

Reply:
No idea.

Reply:
Fixed size in FIF spec

https://github.com/mne-tools/mne-python/blob/master/mne/io/write.py#L362-L387

Reply:
Ah, I should have known 😄 - this should be mentioned in a comment.

Reply:
and we don't expose it as a constant anywhere?

These comments should not be here but wherever this constant is created.
If it is nowhere, I would just create it at the top of this function and call it a day.

```
FIFF_CH_NAME_MAX_LENGTH = 15

...
# we need an extra character since we append '-'
max_n_keep = FIFF_CH_NAME_MAX_LENGTH - 1
n_keep = min(len(ch_stem), max_n_keep - int(np.ceil(np.log10(len(overlaps)))))
```
or

```
# we need an extra character since we append '-'
n_keep = FIFF_CH_NAME_MAX_LENGTH - 1 - int(np.ceil(np.log10(len(overlaps)))
n_keep = min(len(ch_stem), n_keep)
```

whatever makes the code more clear, and the comments shorter